src/transformers/image_box_utils.py

```diff
@@ -0,0 +1,523 @@
+    if check is not None and input_scale == "relative":
+        check_relative_boxes_xyxy(xyxy_boxes, check)
+    elif check is not None and input_scale == "absolute":
+        check_absolute_boxes_xyxy(xyxy_boxes, check)
+
+    # type conversion from intermediate "xyxy" to any output format
+    if output_coords_type == "xywh":
+        converted_boxes = _convert_xyxy_to_xywh(xyxy_boxes)
+    elif output_coords_type == "xcycwh":
+        converted_boxes = _convert_xyxy_to_xcycwh(xyxy_boxes)
+    else:
+        converted_boxes = xyxy_boxes
+
+    # scale conversion (relative -> absolute or absolute -> relative)
+    if input_scale == "relative" and output_scale == "absolute":
+        converted_boxes = _convert_relative_to_absolute(converted_boxes, image_size)
+    elif input_scale == "absolute" and output_scale == "relative":
+        converted_boxes = _convert_absolute_to_relative(converted_boxes, image_size)
+
+    return converted_boxes
+
+
+def _convert_boxes_recursively(
+    boxes: Union[ArrayType, List, Tuple],
+    input_format: str,
+    output_format: str,
+    image_size: Optional[Union[ArrayType, List]] = None,
+    check: Optional[str] = "warn",
+):
+    depth = get_depth_of_nested_objects(boxes)
+
+    # check for "empty" boxes/images
+    if len(boxes) == 0:
+        return boxes
+    elif is_array_type(boxes) and _shape(boxes)[-1] == 0:
+        return boxes
+
+    # check for unsupported depths
+    if depth > 3:
+        raise ValueError(
+            f"Expected boxes to have at most 3 dimensions (n_images, n_boxes, coords), got {depth} dimensions."
+        )
+
+    # check image size is provided correctly
+    if depth == 3 and image_size is not None:
+        image_depth = get_depth_of_nested_objects(image_size)
+        if image_depth == 1:
+            raise ValueError(f"Expected get list of image_sizes but get a single image size: {image_size}.")
+        elif len(image_size) != len(boxes):
+            raise ValueError(
+                f"Expected the same number of images and image sizes, got {len(boxes)} images and {len(image_size)} image sizes."
+            )
+    if depth < 3 and image_size is not None and len(image_size) != 2:
+        raise ValueError(f"Expected image_size to have 2 elements (height, width), got {len(image_size)} elements.")
+
+    # check box has enough coordinates
+    if depth == 1 and len(boxes) < 4:
+        raise ValueError(f"Expected boxes to have at least 4 elements, got {len(boxes)} elements.")
+    if is_array_type(boxes) and _shape(boxes)[-1] < 4:
+        raise ValueError(f"Expected boxes to have at least 4 elements, got {_shape(boxes)[-1]} elements.")
+
+    # Base of recursion.
+    if depth == 2 and is_array_type(boxes):
+        return _convert_boxes_arrays_2d(boxes, input_format, output_format, image_size, check)
+
+    # Recursive approach.
+    elif depth == 1 and isinstance(boxes, (list, tuple) or is_array_type(boxes)):
```

Comment:
Regarding the ` isinstance(boxes, (list, tuple) or is_array_type(boxes))` check - what are the other types `boxes` could be here?